src/softerllama/modeling_softerllama.py

```diff
@@ -224,12 +194,9 @@ def forward(
 
         # upcast attention to fp32
```

Comment:
Why did you upcast to `fp32`, only to downcast back to `query_states.dtype`? Seems redundant

Reply:
Legacy code inherited from original huggingface llama implementation. Doesn't hurt to keep it there, I'm not sure what impact changing this behaviour could cause, and I'm not sure we want to find out either. Apply the age-old adage: If it works, don't touch it.